File: DecompiledGameScripts/cyberpunk/devices/ladders/ladder.swift

public class Ladder extends InteractiveDevice {

  protected cb func OnRequestComponents(ri: EntityRequestComponentsInterface) -> Bool {
    super.OnRequestComponents(ri);
  }

  protected cb func OnTakeControl(ri: EntityResolveComponentsInterface) -> Bool {
    super.OnTakeControl(ri);
    this.m_controller = EntityResolveComponentsInterface.GetComponent(ri, n"controller") as LadderController;
  }

  protected const func HasAnyDirectInteractionActive() -> Bool {
    return true;
  }

  public const func DeterminGameplayRole() -> EGameplayRole {
    return EGameplayRole.OpenPath;
  }

  private const func GetController() -> ref<LadderController> {
    return this.m_controller as LadderController;
  }

  public const func GetDevicePS() -> ref<LadderControllerPS> {
    return this.GetController().GetPS();
  }
}

File: Definitions/Game_Definitions/classes/Ladder.lua
---@meta
---@diagnostic disable

---@class Ladder : InteractiveDevice
Ladder = {}

---@return Ladder
function Ladder.new() return end

---@param props table
---@return Ladder
function Ladder.new(props) return end

---@param ri entEntityRequestComponentsInterface
---@return Bool
function Ladder:OnRequestComponents(ri) return end

---@param ri entEntityResolveComponentsInterface
---@return Bool
function Ladder:OnTakeControl(ri) return end

---@return EGameplayRole
function Ladder:DeterminGameplayRole() return end

---@return LadderController
function Ladder:GetController() return end

---@return LadderControllerPS
function Ladder:GetDevicePS() return end

---@return Bool
function Ladder:HasAnyDirectInteractionActive() return end

